### PR TITLE
refactor(Core/Maps): Change map file version from FourCC to uint32

### PR DESCRIPTION
**Pull request by Kitzunu**
* chery-pick commit <https://github.com/TrinityCore/TrinityCore/commit/42877e75e2a67514dfeb185661e4f397b6eb044a>

Co-Authored-By: Giacomo Pozzoni <giacomopoz@gmail.com>

## Changes Proposed:
-  title
-  Allows to easily increment map version

## Tests Performed:

- none

## How to Test the Changes:

1. Re-extract maps and try to enter game

## Known Issues and TODO List:

- [ ] Maps needs to be re-extracted and uploaded by Shin or Helias

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Winfidonarleyan — APPROVED** [review]

File: src/server/game/Maps/Map.cpp [hunk 1]
```diff
@@ -85,8 +85,12 @@ bool Map::ExistMap(uint32 mapid, int gx, int gy)
         map_fileheader header;
         if (fread(&header, sizeof(header), 1, pf) == 1)
         {
-            if (header.mapMagic != MapMagic.asUInt || header.versionMagic != MapVersionMagic.asUInt)
-                LOG_ERROR("maps", "Map file '%s' is from an incompatible clientversion. Please recreate using the mapextractor.", tmp);
+            if (header.mapMagic != MapMagic.asUInt || header.versionMagic != MapVersionMagic)
+            {
+                LOG_ERROR("maps", "Map file '%s' is from an incompatible map version (%.*u v%u), %.*u v%u is expected. Please pull your source, recompile tools and recreate maps using the updated mapextractor, then replace your old map files with new files.",
```
Comment (Kitzunu) [inline src/server/game/Maps/Map.cpp:90]:
```suggestion
                LOG_ERROR("maps", "Map file '%s' is from an incompatible map version (%.*u v%u), %.*s v%u is expected. Please pull your source, recompile tools and recreate maps using the updated mapextractor, then replace your old map files with new files.",
```

**Comment by FrancescoBorzi** [comment]
what is the status of this PR? needs tests? needs more review? please update the labels @Kitzunu @Winfidonarleyan

**Comment by Winfidonarleyan** [comment]
I think need test this

**Comment by Kitzunu** [comment]
I can't test as mapextractors dont work on windows

**Comment by FrancescoBorzi** [comment]
> I can't test as mapextractors dont work on windows

is there an issue for that? we should fix it!

**Comment by Kitzunu** [comment]
Using With Meshextractors or whatever the cmake option is makes it even worse because then it wont compile :D

**Comment by Kitzunu** [comment]
But someone with linux feel free to test, should be enough if you can start worldserver

**Comment by stoudtlr** [comment]
Was able to test on Windows and it works.  The PR in TrinityCore specifies that mmaps and vmaps do NOT need to be recreated.  Only the maps data.  That extractor does work.  So I only extracted new maps data as stated in that PR.   Was able to start the server without error and logged into game world.
image


Issue with extractors does still need to be looked at but that's a separate issue.

**Comment by Winfidonarleyan** [comment]
@stoudtlr thanks for test pr

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**